python_anvil/api.py

```diff
@@ -85,6 +85,17 @@ def generate_pdf(self, payload: Union[AnyStr, Dict]):
         else:
             raise ValueError("`payload` must be a valid JSON string or a dict")
 
+        # For HTML-type payloads, data must be a dict (or JSON object).
+        if data.type == 'html' and isinstance(data.data, list):
+            raise ValueError(
+                "`payload` data must be a dict when generating a PDF from html"
+            )
+        elif data.type == 'markdown' and isinstance(data.data, dict):
+            raise ValueError(
+                "`payload` data must be a list of dicts when generating a PDF "
+                "from markdown (this is the default)"
+            )
```

Comment:
any final `else` for random `data.type` and/or `data.data` values?

Reply:
I don't think there's any other possibilities due to types already getting validated above this.

Reply:
oh yeah, missed that...of course!